tests/test_interpreter.py

```diff
@@ -8,6 +8,7 @@ def i():
     return finat.indices.DimensionIndex(10)
 
 
+@pytest.mark.xfail
 def test_invalid_binding(i):
```

Comment:
This highlights a problem. When leaving an index sum then the index should not become bound below, it should become actually undefined (ie just not added to the bound_below set). This is a key difference between IndexSum (which is a reduction operation) and ForAll (which is not)